packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -161,13 +188,23 @@ class SharedPreferences {
   }
 
   static Future<Map<String, Object>> _getSharedPreferencesMap() async {
-    final Map<String, Object> fromSystem = await _store.getAll();
+    Map<String, Object> fromSystem;
+    if (_prefixHasBeenSet) {
+      fromSystem = await _store.getAllWithPrefix(_currentPrefix);
```

Comment:
Let's catch unimplemented here, and re-throw an unimplemented with a specific error message saying that they are using an implementation that doesn't support custom prefixes, and that they will need to either not call `setPrefix`, or contact the developer of the implementation for an update. (This will give anyone using a 3P implementation a much less confusing experience.)

Reply:
Is this not already happening with the message I wrote in the platform?

Reply:
Oh, I missed that. I think putting a detailed error message at that layer is problematic because it's relying on knowing things about a higher layer. We want the error message to be plugin-client terms, which means app-facing API terms. What if we change `shared_preferences` to not have `setPrefix` any more, for instance? Or what if we change `shared_preferences` down the road to drop the older codepath and always use `getAllWithPrefix`, and now every call `getInstance` while using an older implementation says not to call `setPrefix`, which they haven't done?

I would much rather we keep the app-developer-facing error message in the app-facing package.